src/ol/render/canvas/Executor.js

```diff
@@ -47,6 +47,20 @@ const tmpExtent = createEmpty();
 const tmpTransform = createTransform();
 
 
+let measureFont, measureScale, measureWidths;
+/**
+ * @param {string} text Text.
+ * @return {number} Measured width, using `measureFont`, `measureScale` and `measureWidths`.
+ */
+function measure(text) {
+  let width = measureWidths[text];
+  if (!width) {
+    width = measureWidths[text] = measureTextWidth(measureFont, text);
+  }
```

Comment:
Do we ever get `0` from `measureTextWidth`?

Might be nicer to
```js
if (!(text in measuredWidths)) {
    measuredWidths[text] = measureTextWidth(measureFont, text);
}
return measuredWidths[text] * measureScale;
```

Or with `let width` to avoid the second lookup.